core/toolbox/category.js

```diff
@@ -194,6 +194,147 @@ Blockly.ToolboxCategory.prototype.parseContents_ = function(categoryDef,
   return toolboxItems;
 };
 
+/**
+ * @override
+ */
+Blockly.ToolboxCategory.prototype.createDom = function() {
+  this.htmlDiv_ = this.createContainer_();
+  Blockly.utils.aria.setRole(this.htmlDiv_, Blockly.utils.aria.Role.TREEITEM);
+  Blockly.utils.aria.setState(/** @type {!Element} */ (this.htmlDiv_),
+      Blockly.utils.aria.State.SELECTED,false);
+  Blockly.utils.aria.setState(/** @type {!Element} */ (this.htmlDiv_),
+      Blockly.utils.aria.State.LEVEL, this.level_);
+  this.htmlDiv_.setAttribute('id', this.id_);
+
+  this.rowDiv_ = this.createRowContainer_();
+  this.htmlDiv_.appendChild(this.rowDiv_);
+
+  this.iconSpan_ = this.createIconSpan_();
+  Blockly.utils.aria.setRole(this.iconSpan_, Blockly.utils.aria.Role.PRESENTATION);
```

Comment:
Can you set roles inside your helper functions, at least for this, the label, and the group?

Reply:
I went back and forth with this a bit. By keeping these in the `createDom` method someone can overwrite the create methods without losing the accessibility labels. If we put them in their individual create methods we lose this, but increase the ability for the developer to create their own accessibility labels. Thoughts?

Reply:
Do we expect all changes to the create methods to end up needing the same accessibility labels?

I see your point, though. There's a balance between code structure and making sure that accessibility is the default, and I think the latter is more important here.

If the accessibility labels end up being more than a few lines, let's put them in helper methods going forward.

Reply:
Yeah, I think it depends a lot on the use case. They would definitely need to change the labels if they go away from the tree structure, but hard to say otherwise.

sgtm